auth-jwt/Sources/App/Middleware/BearerAuthenticator.swift

```diff
@@ -0,0 +1,45 @@
+import Foundation
+import Hummingbird
+import HummingbirdAuth
+import JWTKit
+
+struct TokenPayload: HBAuthenticatable {
+  var subject:String
+}
+
+protocol DataProtocol {}
+
+struct BearerAuthenticator: HBAsyncAuthenticator, DataProtocol {
+  var jwksUrl: String
+
+  func authenticate(request: HBRequest) async throws -> TokenPayload? {
+    guard let jwtToken = request.authBearer?.token else { throw HBHTTPError(.unauthorized) }
+    
+    struct TestPayload: JWTPayload, Equatable {
+      enum CodingKeys: String, CodingKey {
+        case subject = "sub"
+        case expiration = "exp"
+      }
+
+      var subject: SubjectClaim
+      var expiration: ExpirationClaim
+
+      func verify(using signer: JWTSigner) throws {
+        try self.expiration.verifyNotExpired()
+      }
+    }
+
+    let jwksData = try Data(
```

Comment:
Should we load the jwks url in the initialiser for the middleware, instead of every time you call the middleware?